src/modules/ce/components/AssignContactFormItem.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useCallback, useMemo } from 'react';
+import FormSelect from '@/modules/form/components/FormSelect';
+import {
+  CeReferralSwimlaneFieldsFragment,
+  PickListOption,
+  PickListType,
+  useGetPickListQuery,
+} from '@/types/gqlTypes';
+
+interface Props {
+  swimlane: CeReferralSwimlaneFieldsFragment;
+  users: string[];
+  setUsers: (userIds: string[]) => void;
+}
+const AssignContactFormItem: React.FC<Props> = ({
+  swimlane,
+  users,
+  setUsers,
+}) => {
+  const {
+    data: { pickList: staffPickList } = {},
+    loading: staffPickListLoading,
+    error: staffPickListError,
+  } = useGetPickListQuery({
+    variables: {
+      pickListType: PickListType.Users,
+    },
+  });
+
+  const selectedUsers = useMemo(
+    () =>
+      users.map((userId) => {
+        return { code: userId };
```

Comment:
With the approach of only tracking `id`s, if the user is not in the picklist then they will appear as just a selected ID. Which the user won't be able to understand. Seems OK for now, but the approach may need to change depending on how we want to handle users who were assigned to a swimlane or task, then became inactive or lost permission to participate in referrals. Ticket is https://github.com/open-path/Green-River/issues/7615  maybe we can add a note about this in the code?

Reply:
thanks, yeah I think that is a nice improvement!